task-assigning/task-assigning-service/src/main/java/org/kie/kogito/taskassigning/service/TaskAssigningService.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.taskassigning.service;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.quarkus.runtime.Startup;
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.kie.kogito.taskassigning.core.model.TaskAssigningSolution;
+import org.kie.kogito.taskassigning.service.config.TaskAssigningConfig;
+import org.kie.kogito.taskassigning.service.config.TaskAssigningConfigValidator;
+import org.kie.kogito.taskassigning.user.service.api.UserServiceConnector;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class in experimental status! don't waste time here!
+ * The only objective by now is to be sure the tasks can be consumed and the solver started.
+ */
+@ApplicationScoped
+@Startup
+public class TaskAssigningService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskAssigningService.class);
+
+    @Inject
+    SolverFactory<TaskAssigningSolution> solverFactory;
+
+    @Inject
+    TaskAssigningConfig config;
+
+    @Inject
+    ManagedExecutor managedExecutor;
+
+    @Inject
+    TaskServiceConnector taskServiceConnector;
+
+    @Inject
+    UserServiceConnector userServiceConnector;
+
+    SolverExecutor solverExecutor;
+
+    SolutionDataLoader solutionDataLoader;
+
+    AtomicReference<String> serviceStatus = new AtomicReference<>("Stopped");
+
+    int totalChances = 3;
+
+    @PostConstruct
+    void start() {
+        serviceStatus.set("Starting");
+        startUpValidation();
+        solverExecutor = new SolverExecutor(solverFactory, solution -> {
+            serviceStatus.set("Solution produced!");
+            LOGGER.debug("A new solution has been produced {}", solution);
+        });
+
+        managedExecutor.execute(solverExecutor);
+        solutionDataLoader = new SolutionDataLoader(taskServiceConnector,
+                                                    userServiceConnector,
+                                                    Duration.ofMillis(5000));
```

Comment:
Long literals should take the suffix "L" to make the type obvious.

```suggestion
                                                    Duration.ofMillis(5000L));
```

Reply:
This complete TaskAssigningServiceTask is right now just an experimental thing for putting the other pieces working together,  don't suffer here, this duration will come from the application configuration values.